specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -2064,93 +2941,269 @@
+        "keyVaultProperties"
+      ],
+      "type": "object",
+      "properties": {
+        "keyVaultProperties": {
+          "$ref": "#/definitions/EncryptionKeyVaultUpdateProperties"
+        }
+      }
+    },
+    "EndpointDeploymentModel": {
+      "type": "object",
+      "properties": {
+        "format": {
+          "description": "Model format",
+          "type": "string"
+        },
+        "name": {
+          "description": "Model name.",
+          "type": "string"
+        },
+        "source": {
+          "description": "Optional. Deployment model source ARM resource ID.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Model version.",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointDeploymentResourceProperties": {
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "failureReason": {
+          "description": "The failure reason if the creation failed.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Read-only provision state status property.",
+          "$ref": "#/definitions/DefaultResourceProvisioningState"
+        },
+        "type": {
+          "description": "Kind of the deployment.",
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9._]",
+          "type": "string"
+        }
+      },
+      "discriminator": "type"
+    },
+    "EndpointDeploymentResourcePropertiesBasicResource": {
+      "required": [
+        "properties"
       ],
-      "type": "string",
-      "x-ms-enum": {
-        "name": "DiagnoseResultLevel",
-        "modelAsString": true
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EndpointDeploymentResourceProperties"
+        }
       }
     },
-    "DiagnoseWorkspaceParameters": {
-      "description": "Parameters to diagnose a workspace",
+    "EndpointDeploymentResourcePropertiesBasicResourceArmPaginatedResult": {
       "type": "object",
       "properties": {
+        "nextLink": {
+          "type": "string"
+        },
         "value": {
-          "$ref": "#/definitions/DiagnoseRequestProperties"
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+          }
         }
       }
     },
-    "EncryptionKeyVaultUpdateProperties": {
-      "required": [
-        "keyIdentifier"
-      ],
+    "EndpointKeys": {
       "type": "object",
       "properties": {
-        "keyIdentifier": {
-          "minLength": 1,
-          "pattern": "[a-zA-Z0-9_]",
+        "keys": {
+          "description": "Dictionary of Keys for the endpoint.",
+          "type": "object",
+          "x-ms-secret": true
+        }
```

Comment:
free form objects are not allowed . Please define these with exact schema not just as "object"

Reply:
removed and use the account API key schema